src/ts/chat/stores/ChatStore.ts

```
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import ChatActions from '../actions/ChatActions';
+import CSEChat from '../util/CSEChat';
+import Config from '../util/Config';
+import * as React from 'react';
+import * as Reflux from 'reflux';
+import messageType from '../constants/messageType';
+
+const ChatStore : any = Reflux.createStore({
+  name: 'ChatStore',
+  chat: <CSEChat> null,
+  connected: false,
+  updated: 0,
+  errorListener: null,
+
+  listenables: [ ChatActions ],
+
+  // ChatAction.connect(...)
+  onConnect: function(
+    username:string|(()=>string), 
+    password:string|(()=>string), 
+    nick:string, 
+    rooms:string[] = [ '_global', '_it', '_modsquad' ])
+  {
+    this.connected = false;
+    this.updated = 0;
+    this.config = new Config(username, password, nick);
+
+    this.connect(rooms);
+  },
+
+  connect: function(rooms: string[]) {
+
+    this.chat = new CSEChat(this.config);
```

Would you prefer silently ignoring the extra call to connect or it report an error.   If the latter, should probably do the same where we check for being connected, e.g. in the send message methods.

Personally, I would rather it throw an error, than silently ignore.